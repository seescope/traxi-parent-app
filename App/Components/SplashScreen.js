/* eslint global-require: "off" */

import React from 'react';
import { Dimensions, View, Text, Image } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { connect } from 'react-redux';
import * as Animatable from 'react-native-animatable';

import Background from '../Components/Background';
import Button from '../Components/Button';
import Spacing from '../Components/Spacing';
import { WHITE, TRANSPARENT } from '../Constants/Colours';
import loginWithMethod from '../Actions/LoginWithMethod';
import { logError } from '../Utils';

const { height } = Dimensions.get('window');
const handleError = error => {
  logError(`Error logging in: ${error}`);
  alert('There was an error logging you in. Please try again.');
};

const logoStyle = {
  backgroundColor: TRANSPARENT,
  color: WHITE,
  fontSize: 28,
  textAlign: 'center',
  fontFamily: 'Raleway-ExtraBold',
  marginBottom: 16,
};

const containerStyle = {
  flex: 1,
  alignItems: 'center',
  paddingHorizontal: 32,
};

const imageStyle = {
  height: height / 2,
};

const SplashScreen = () =>
  <Background style={containerStyle}>
    <Animatable.Image
      useNativeDriver
      resizeMode="contain"
      style={imageStyle}
      animation="bounceInDown"
      source={require('../Images/trail_preview.png')}
    />

    <Spacing height={32} />

    <Animatable.Text
      useNativeDriver
      animation="bounceIn"
      delay={1000}
      style={logoStyle}
    >
      Start seeing what your kids are doing.
    </Animatable.Text>

    <Spacing height={32} />

    <Animatable.View
      useNativeDriver
      animation="bounceInUp"
      delay={2000}
    >
      <Button onPress={() => Actions.intro()}>Show me how</Button>
    </Animatable.View>
  </Background>;


SplashScreen.propTypes = {
  getStarted: React.PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  getStarted: () => dispatch(loginWithMethod())
    .then(() => Actions.setName())
    .catch(handleError),
});

export default connect(null, mapDispatchToProps)(SplashScreen);
