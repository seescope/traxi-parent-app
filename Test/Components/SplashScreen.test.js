import React from 'react';
import renderer from 'react-test-renderer';
import SplashScreen from '../../App/Components/SplashScreen';


it('renders the <SplashScreen> component', () => {
  const tree = renderer
    .create(<SplashScreen />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
