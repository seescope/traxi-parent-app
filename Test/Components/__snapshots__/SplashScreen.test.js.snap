exports[`test renders the <SplashScreen> component 1`] = `
<View
  colors={
    Array [
      "#448AFF",
      "#90CAF9",
    ]
  }
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingHorizontal": 32,
    }
  }>
  <Image
    animation="bounceInDown"
    source={1} />
  <View
    style={
      Object {
        "height": 32,
      }
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    animation="bounceIn"
    delay={1000}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#FFF",
        "fontFamily": "Raleway-ExtraBold",
        "fontSize": 32,
        "marginBottom": 16,
        "marginHorizontal": 16,
        "textAlign": "center",
      }
    }>
    Start seeing what your kids are doing.
  </Text>
  <View
    style={
      Object {
        "height": 32,
      }
    } />
  <View
    animation="bounceInUp"
    delay={2000}>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFF",
            "fontFamily": "Raleway-Regular",
            "fontSize": 20,
          }
        }>
        Show me how
         
      </Text>
    </View>
  </View>
</View>
`;
